### ISSUE TITLE
Gluster Volume per Container or Shared Gluster Volume

### ISSUE THREAD

**Issue by Vad1mo** [sapk/docker-volume-gluster#14]
I manually created a gluster volume `gv0`.  Now I had the expectation that when use the plugin, I can use this single gluster volume `gv0` for multiple container. 

I guess It is not intended to be used in this way, (or I can't find an alternative), because the data are stored in the root folder of the gluster volume `gv0/`. I was actually expecting them in `gv0/portainer-data/...` 

So is it intended that each container should have its own gluster volume? 

```
volumes:
  portainer-data:
    driver: glusterfs
    driver_opts:
      voluri: "localhost:gv0"
```

**Comment by sapk** [comment]
That how docker-compose works. It define the name of volume by default by prepending with stack name. You can define a volume in docker and use docker-compose volume external : https://docs.docker.com/compose/compose-file/#external. If you use plugin directly in host not docker plugin it has a option to mount gluster on gluster volume name so even if there is multiple docker volume only on is mount and shared by gluster. https://github.com/sapk/docker-volume-gluster/blob/85f64e407d486e42a24b2064eecf5a8a7735f7e5/gluster/gluster.go#L19. It should be able to do the same with docker-plugin but need to be passed by env variable (not implemented)

**Comment by Vad1mo** [comment]
the name of the volume `glusterfs:latest    portainer_portainer-data` is fine for me. I was expecting that there is a way where its possible to use a gluster volume like `gv0` for many different containers at the same time. eg. `gv0/postgress-db-data` and `gv0/mysql-db-dta` 

But I guess the best option for that is to automatically create gluster volumes on demand.

**Comment by Vad1mo** [comment]
btw. did you look at https://github.com/heketi/heketi, its basically a rest api for gluster, might be an option if you plan to add an option that can create volumes on demand

**Comment by sapk** [comment]
If you use --mount-uniq it will mount base on volume name so only one mount on host.

**Comment by sapk** [comment]
yes in the todo list #5 and add env #15 fro docker-plugin jsut rigth now ^^

**Comment by sapk** [comment]
fix by #16
